tests/unit/tools/test_tools_utils.py

```diff
@@ -149,3 +149,240 @@ def test_authorlist_ignores_space_between_authors_and_affiliations():
+        '1\n'
+        'K. Agashe\n'
+        '2\n'
+        'G. Aielli\n'
+        '1,\n'
+        '2\n'
+        '1\n'
+        'Universit`a di Bologna and INFN, Dip. Scienze per la '
+        'Qualit`a della Vita, I-47921, Rimini, Italy\n'
+        '2\n'
+        'University of Maryland, Department of Physics, College '
+        'Park, MD 20742-4111, USA'
+    )
+
+    expected = (
+        '100__ $$aPatrignani, C.$$vUniversità di Bologna and INFN, Dip. '
+        'Scienze per la Qualità della Vita, I-47921, Rimini, Italy\n'
+        '700__ $$aAgashe, K.$$vUniversity of Maryland, Department of Physics, '
+        'College Park, MD 20742-4111, USA\n'
+        '700__ $$aAielli, G.$$vUniversità di Bologna and INFN, Dip. Scienze '
+        'per la Qualità della Vita, I-47921, Rimini, Italy $$vUniversity of '
+        'Maryland, Department of Physics, College Park, MD 20742-4111, USA'
+    )
+
+    result = authorlist(text)
+
+    assert expected == result.encode('utf-8')
+
+def test_authorlist_affids_with_dots():
+    text = (
+        'C. Patrignani\n'
+        '1,\n'
+        'K. Agashe\n'
+        '2,\n'
+        'G. Aielli\n'
+        '1,\n'
+        '2\n'
+        '1.\n'
+        'Universit`a di Bologna and INFN, Dip. Scienze per la '
+        'Qualit`a della Vita, I-47921, Rimini, Italy\n'
+        '2.\n'
+        'University of Maryland, Department of Physics, College '
+        'Park, MD 20742-4111, USA'
+    )
+
+    expected = (
+        '100__ $$aPatrignani, C.$$vUniversità di Bologna and INFN, Dip. '
+        'Scienze per la Qualità della Vita, I-47921, Rimini, Italy\n'
+        '700__ $$aAgashe, K.$$vUniversity of Maryland, Department of Physics, '
+        'College Park, MD 20742-4111, USA\n'
+        '700__ $$aAielli, G.$$vUniversità di Bologna and INFN, Dip. Scienze '
+        'per la Qualità della Vita, I-47921, Rimini, Italy $$vUniversity of '
+        'Maryland, Department of Physics, College Park, MD 20742-4111, USA'
+    )
+    result = authorlist(text)
+
+    assert expected == result.encode('utf-8')
+
+
+def test_authorlist_no_commas_between_affids():
+    text = (
+        'C. Patrignani\n'
+        '1,\n'
+        'K. Agashe\n'
+        '2,\n'
+        'G. Aielli\n'
+        '1\n'
+        '2\n'
+        '1.\n'
+        'Universit`a di Bologna and INFN, Dip. Scienze per la '
+        'Qualit`a della Vita, I-47921, Rimini, Italy\n'
+        '2.\n'
+        'University of Maryland, Department of Physics, College '
+        'Park, MD 20742-4111, USA'
+    )
+
+    with pytest.raises(KeyError):
+        authorlist(text)
+
+def test_authorlist_multiple_affiliations_on_single_line():
+    text = (
+        'A. Aduszkiewicz\n'
+        '1\n'
+        ', Y.X. Ali\n'
+        '1,20\n'
+        ', E I Andronov\n'
+        '20\n'
+        ', Einstein\n'
+        '13,15\n'
+        '\n'
+        '1 CERN\n'
+        '20 DESY\n'
+        '13 ETH ZÜRICH15PRINCETON\n'
+    )
+
+    with pytest.raises(KeyError):
+        authorlist(text)
```

Comment:
It's not enough to test that it raises, you should also assert here that the exception message is understandable:
```python
with pytest.raises(KeyError) as excinfo:
    authorlist(text)
assert 'Multiple affiliations on a single line' in str(excinfo.value)
```
(typed from memory, might contain errors).